File: backend/internal/models/team.go
package models

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type TeamMemberModel struct {
	ID          primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
	Name        string             `bson:"name" json:"name"`
	Position    map[string]string  `bson:"position" json:"position"`
	Description map[string]string  `bson:"description" json:"description"`
	Email       string             `bson:"email" json:"email"`
	Linkedin    string             `bson:"linkedin" json:"linkedin"`
}

type TeamModel struct {
	ID         primitive.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
	Period     string             `bson:"period" json:"period"`
	Management []TeamMemberModel  `bson:"management" json:"management"`
	Auditing   []TeamMemberModel  `bson:"auditing" json:"auditing"`
}

File: frontend/src/types/Team.ts
export interface TeamMember {
    _id?: string;
    name: string;
    position: Record<string, string>;
    description: Record<string, string>;
    email: string;
    linkedin: string;
}

export interface Team {
    _id?: string;
    period: string;
    management: TeamMember[];
    auditing: TeamMember[];
}
